### PR TITLE
chore(server): refactor yaml generation

### PR DESCRIPTION
**Pull request by schoren**
This PR refactors yaml file generation to remove coupling with openapi models.

## Changes

- Remove yaml coupling with openapi models
-  Integrate a DeepCopy library to simplify the converting process.

## Fixes

- cli env injector not handling pointers to structs

## Checklist

- [x] tested locally
- [x] added new dependencies
- [ ] updated the docs
- [x] added a test

### CODE REVIEW

File: server/model/yaml/http.go [hunk 1]
```diff
@@ -164,6 +124,7 @@ type HTTPAPIKeyAuth struct {
 }
 
 func (aka HTTPAPIKeyAuth) Validate() error {
+	fmt.Println("ACA!!!!", aka.Value)
```
Comment (mathnogueira) [inline server/model/yaml/http.go:127]: You forgot to remove this

File: server/http/mappings/http.go [hunk 2]
```diff
@@ -112,28 +112,28 @@ func (m Model) HTTPResponse(in openapi.HttpResponse) *model.HTTPResponse {
...
+	// 	props = map[string]string{
+	// 		"token": in.Bearer.Token,
+	// 	}
+	// }
 
 	return &model.HTTPAuthenticator{
-		Type:  in.Type,
-		Props: props,
+		Type: in.Type,
+		// Props: props,
```
Comment (mathnogueira) [inline server/http/mappings/http.go:137]: Don't you have to map the arguments for each auth type?

**Review by mathnogueira — APPROVED** [review]
GJ, just a question and a possible debug log you forgot to remove

File: server/http/mappings/http.go (hunk 2 above)
Comment (schoren) [reply server/http/mappings/http.go:137]: uppps, forgot to fix this. Updated

**Review by mathnogueira — APPROVED** [review]

**Pull request merged**